src/librustc_span/lib.rs

```diff
@@ -1044,9 +1044,7 @@ impl Decodable for SourceFile {
                 name_was_remapped,
                 unmapped_path: None,
                 // `crate_of_origin` has to be set by the importer.
```

Comment:
If we are using `CrateNum` we should be able to just encode and decode it here in `librustc_span/lib.rs` without help of `fn imported_source_files` in `decoder.rs`.